Include/internal/pycore_global_objects.h

```diff
@@ -34,6 +34,31 @@ extern "C" {
     }
 
 
+/* bytes objects */
+
+#define _PyBytes_EMPTY_INIT \
+    { \
+        _PyVarObject_IMMORTAL_INIT(&PyBytes_Type, 0), \
+        .ob_shash = -1, \
+        .ob_sval[0] = '\0', \
+    }
+#define _PyBytes_CHAR_INIT(CH) \
+    { \
+        { \
+            _PyVarObject_IMMORTAL_INIT(&PyBytes_Type, 1), \
+            .ob_shash = -1, \
+            .ob_sval = { CH }, \
+        }, \
+    }
+
+static inline void
+_PyBytes_reset(PyBytesObject *op)
+{
+    // Force a new hash to be generated since the hash seed may have changed.
```

Comment:
When and how can the hash seed change?

Reply:
It looks like one *can* change the hash seed (using the recipe Eric gave) but it will break things and it's unclear that it's desirable. I can't for the life of me figure out exactly on which code path from Py_Initialize() the random hash seed is set (I see the code but not how it's being called),  but I can't prove that it won't change in the above scenario.

Eric, can you confirm that the above scenario results in a different hash seed? That would be bad indeed.

Reply:
I think the following is the status quo:

1. early in runtime init (via `_Py_HashRandomization_Init()` in Python/bootstrap_hash.c):
   * the hash secret (`_Py_HashSecret`) is set using the configured (`PyConfig`) hash seed (or none)
   * `_Py_HashSecret_Initialized` is set to true
2. during runtime fini neither `_Py_HashSecret` nor `_Py_HashSecret_Initialized` is modified
3. if the runtime is initialized again, `_Py_HashSecret_Initialized` is still true and any requested hash seed is ignored (so `_Py_HashSecret` is left alone)

So I got that wrong.  I missed that `_Py_HashSecret_Initialized` is not cleared during finalization (step 2).  Only the configuration during the first time the runtime is initialized is respected.

> on which code path from Py_Initialize() the random hash seed is set

```
Py_Initialize() -> Py_InitializeEx() -> Py_InitializeFromConfig()

Py_InitializeFromConfig() -> pyinit_core() -> pyinit_config() -> pycore_init_runtime() -> _Py_HashRandomization_Init()
```

Reply:
...and sorry about the confusion.  I should have double-checked.